xds/src/main/java/io/grpc/xds/AbstractXdsClient.java

```diff
@@ -268,6 +276,10 @@ protected final void nackResponse(ResourceType type, String nonce, String errorD
     adsStream.sendDiscoveryRequest(type, versionInfo, resources, nonce, errorDetail);
   }
 
+  protected final SynchronizationContext getSyncContext() {
```

Comment:
The `syncContext` is used through-out with 2 common patterns:
- `execute` with. a Runnable
- ` schedule` with a few params and a `ScheduledExecutorService` which is always `getTimeService()`

Have you considered just providing 2 methods in `AbstractXdsClient` that allow the caller to do the same thing but without calling `getSyncContext()`? So the `executeSynchronously` is just `getSyncContext().execute()` and `scheduleSynchronously` is just  `getSyncContext().schedule()` with the `ScheduledExecutorService` always `getTimeService()`. This will simplify a lot of boilerplate and also hide the actual implementation of syncContext and timeService.

Reply:
`getSyncContext().execute()` is NOT executing it synchronously.

We could have `AbstractXdsClient` provide simpler method, such as `execute(Runnable)` and `schedule(long, TimeUnit)`, and hide the implementation detail of how it is actually scheduled into the SynchronizationContext. But that isn't significantly simpler, it doesn't bother doing that.

Reply:
okay.